### ISSUE TITLE
[Feature Request] Allow StaticIntTuple mask for SIMD.shuffle

### ISSUE THREAD

**Issue by mikowals** [modularml/mojo#2293]
### Review Mojo's priorities

- [X] I have read the [roadmap and priorities](https://docs.modular.com/mojo/roadmap.html#overall-priorities) and I believe this request falls within the priorities.

### What is your request?

As title.  I want to build the list of indices in the mask with a for-loop at compile time.  StaticIntTuple seems like the obvious candidate for such a mask but I other types could work.

I already implemented this and can finish the PR after inferred parameters are in nightly.

### What is your motivation for this change?

Reduce boilerplate writing overload functions for different mask sizes.  See Discord [discussion](https://discord.com/channels/1087530497313357884/1205809638751346688/1205817726086418463).

### Any other details?

The current `SIMD.shuffle` relies on `_shuffle_list` with the signature:

```mojo
fn _shuffle_list[ *mask: Int, output_size: Int = size ](self, other: Self) -> SIMD[type, output_size]:
```

For this feature the signature of `_shuffle_list` using `StaticIntTuple` would be like this:

```mojo
fn _shuffle_list[inferred output_size: Int, mask: StaticIntTuple[output_size]](self, other: Self) -> SIMD[type, output_size]:
```

Getting these to have the order of `output_size` and `mask` the same would be nicer.  For most cases it would be hidden behind `SIMD.shuffle`.

**Comment by LJ-9801** [comment]
Any news on when the inferred parameters will be supported? Seems like a great feature to have.

**Comment by mikowals** [comment]
According to [this](https://github.com/modularml/mojo/issues/1245) it is done in main branch so it should be in nightly branch soon and released this month or next I suppose.

**Comment by mikowals** [comment]
And if you are keen for this shuffle functionality the implementation is trivial.

```
@always_inline("nodebug")
fn _shuffle_list[
    dtype: Dtype, size: Int, output_size: Int, mask: StaticIntTuple[output_size]
](a: SIMD[dtype: size], other: SIMD[dtype, size]) -> SIMD[dtype, output_size]:
    
    #could check that each entry in mask is greater than or equal to 0 and less than 2 * size
    return __mlir_op.`pop.simd.shuffle`[
        mask = mask.data.array,
        _type = __mlir_type[
            `!pop.simd<`, output_size.value, `, `, type.value, `>`
        ],
    ](a.value, b.value)
```

**Comment by mikowals** [comment]
fixed by #2315
